external-effects/super-blur/CMakeLists.txt

```diff
@@ -0,0 +1,161 @@
+	add_definitions( -DUSE_IMAGEMAGICK=1 )
+	SET(CMAKE_SWIG_FLAGS "-DUSE_IMAGEMAGICK=1")
+
+ENDIF (ImageMagick_FOUND)
+
+################### FFMPEG #####################
+# Find FFmpeg libraries (used for video encoding / decoding)
+FIND_PACKAGE(FFmpeg REQUIRED)
+
+foreach(ffmpeg_comp AVCODEC AVDEVICE AVFORMAT AVFILTER AVUTIL POSTPROC SWSCALE SWRESAMPLE AVRESAMPLE)
+    if(${ffmpeg_comp}_FOUND)
+        # Include FFmpeg headers (needed for compile)
+        list(APPEND FF_INCLUDES ${${ffmpeg_comp}_INCLUDE_DIRS})
+        add_definitions(${${ffmpeg_comp}_DEFINITIONS})
+    endif()
+endforeach()
+list(REMOVE_DUPLICATES FF_INCLUDES)
+include_directories(${FF_INCLUDES})
+
+################# LIBOPENSHOT-AUDIO ###################
+# Find JUCE-based openshot Audio libraries
+FIND_PACKAGE(OpenShotAudio 0.1.8 REQUIRED)
+
+# Include Juce headers (needed for compile)
+include_directories(${LIBOPENSHOT_AUDIO_INCLUDE_DIRS})
+
+################# QT5 ###################
+# Find QT5 libraries
+foreach(qt_lib Qt5Widgets Qt5Core Qt5Gui Qt5Multimedia Qt5MultimediaWidgets)
+    find_package(${qt_lib} REQUIRED)
+    # Header files
+    list(APPEND QT_INCLUDES ${${qt_lib}_INCLUDE_DIRS})
+    # Compiler definitions
+    add_definitions(${${qt_lib}_DEFINITIONS})
+    # Other CFLAGS
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${${qt_lib}_EXECUTABLE_COMPILE_FLAGS}")
+endforeach()
+list(REMOVE_DUPLICATES QT_INCLUDES)
+include_directories(${QT_INCLUDES})
+
+# Manually moc Qt files
+qt5_wrap_cpp(MOC_FILES ${QT_HEADER_FILES})
+
+
+################# BLACKMAGIC DECKLINK ###################
+IF (ENABLE_BLACKMAGIC)
+	# Find BlackMagic DeckLinkAPI libraries
+	FIND_PACKAGE(BlackMagic)
+
+	IF (BLACKMAGIC_FOUND)
+		# Include Blackmagic headers (needed for compile)
+		include_directories(${BLACKMAGIC_INCLUDE_DIR})
+	ENDIF (BLACKMAGIC_FOUND)
+ENDIF (ENABLE_BLACKMAGIC)
+
+################### OPENMP #####################
+# Check for OpenMP (used for multi-core processing)
+FIND_PACKAGE(OpenMP)
+
+if (OPENMP_FOUND)
+   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS} ")
+endif(OPENMP_FOUND)
+
+################### ZEROMQ #####################
+# Find ZeroMQ library (used for socket communication & logging)
+FIND_PACKAGE(ZMQ REQUIRED)
+
+# Include ZeroMQ headers (needed for compile)
+include_directories(${ZMQ_INCLUDE_DIRS})
+
+################### RESVG #####################
+# Find resvg library (used for rendering svg files)
+FIND_PACKAGE(RESVG)
+
+# Include resvg headers (optional SVG library)
+if (RESVG_FOUND)
+	include_directories(${RESVG_INCLUDE_DIRS})
+endif(RESVG_FOUND)
+
+################### JSONCPP #####################
+# Include jsoncpp headers (needed for JSON parsing)
+if (USE_SYSTEM_JSONCPP)
+	find_package(JsonCpp REQUIRED)
+	include_directories(${JSONCPP_INCLUDE_DIRS})
+else()
+	include_directories("../../thirdparty/jsoncpp")
+endif(USE_SYSTEM_JSONCPP)
+
+SET ( EFFECT_SOURCE_FILES
+		SuperBlur.cpp)
+
+IF (NOT DISABLE_EXTERNAL_EFFECTS)
```

Comment:
This check should go into the `external-effects/CMakeLists.txt` file instead, so the `add_subdirectory(super-blur)` is never even executed when external effects are disabled.

Reply:
(In fact, it could also go into the root CMakeLists.txt, and prevent the `add_subdirectory(external-effects)` from even happening. I have no strong preference on that one. But it should definitely be wrapped around an `add_subdirectory()` call somewhere, so that CMake never even descends into that dir if there's ultimately nothing for it to do there.

Reply:
Done